[bun i] scripts in child workspaces pastes x2 "\node_modules" in resolved path. which is unwanted/bug behaviour

### What version of Bun is running?

1.1.18-canary.1+ee2561819

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

repro repo: https://github.com/sequencerr/bun-repro-workspaces/tree/postinstall
run - `bun i` in project root

p.s.: start of conversation https://discord.com/channels/876711213126520882/1149339379446325248/1259179223130837002

### What is the expected behavior?

to successfully run `node D:\***\bun-repro-postinstall\node_modules\puppeteer\lib\esm\puppeteer\node\cli.js ...` command in postinstall package.json script

### What do you see instead?

```
PS D:\***\bun-repro-postinstall> bun i
bun install v1.1.18-canary.1 (ee256181)
  ⚙️  asd [1/2] node:internal/modules/cjs/loader:1147
  throw err;
  ^

Error: Cannot find module 'D:\***\bun-repro-postinstall\node_modules\node_modules\puppeteer\lib\esm\puppeteer\node\cli.js'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
    at Module._load (node:internal/modules/cjs/loader:985:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v20.11.0

error: postinstall script from "asd" exited with 1
```

### Additional information

_No response_

Hm... I have similar issue with bun v1.1.18 on Windows: `Cannot find module 'W:\W:\web\…`. Drive letter is added twice for some reason. No workspaces though.

--- a/src/bun.js/api/bun/process.zig
+++ b/src/bun.js/api/bun/process.zig
@@ -1513,8 +1513,10 @@ pub fn spawnProcessWindows(
     const allocator = stack_allocator.get();
     const loop = options.windows.loop.platformEventLoop().uv_loop;
 
-    const cwd = try allocator.dupeZ(u8, options.cwd);
-    defer allocator.free(cwd);
+    var cwd_buf: bun.PathBuffer = undefined;
+    @memcpy(cwd_buf[0..options.cwd.len], options.cwd);
+    cwd_buf[options.cwd.len] = 0;
+    const cwd = cwd_buf[0..options.cwd.len :0];
 
     uv_process_options.cwd = cwd.ptr;
 

--- a/src/install/lifecycle_script_runner.zig
+++ b/src/install/lifecycle_script_runner.zig
@@ -106,7 +106,7 @@ pub const LifecycleScriptSubprocess = struct {
 
         const manager = this.manager;
         const original_script = this.scripts.items[next_script_index].?;
-        const cwd = bun.path.z(this.scripts.cwd, &cwd_z_buf);
+        const cwd = this.scripts.cwd;
         const env = manager.env;
         this.stdout.setParent(this);
         this.stderr.setParent(this);

--- a/src/install/lockfile.zig
+++ b/src/install/lockfile.zig
@@ -2767,7 +2767,7 @@ pub const Package = extern struct {
             items: [Lockfile.Scripts.names.len]?Lockfile.Scripts.Entry,
             first_index: u8,
             total: u8,
-            cwd: string,
+            cwd: stringZ,
             package_name: string,
 
             pub fn printScripts(
@@ -2955,7 +2955,7 @@ pub const Package = extern struct {
             this: *const Package.Scripts,
             lockfile: *Lockfile,
             lockfile_buf: []const u8,
-            cwd: string,
+            cwd_: string,
             package_name: string,
             resolution_tag: Resolution.Tag,
             add_node_gyp_rebuild_script: bool,
@@ -2963,11 +2963,26 @@ pub const Package = extern struct {
             const allocator = lockfile.allocator;
             const first_index, const total, const scripts = getScriptEntries(this, lockfile, lockfile_buf, resolution_tag, add_node_gyp_rebuild_script);
             if (first_index != -1) {
+                var cwd_buf: if (Environment.isWindows) bun.PathBuffer else void = undefined;
+
+                const cwd = if (comptime !Environment.isWindows)
+                    cwd_
+                else brk: {
+                    @memcpy(cwd_buf[0..cwd_.len], cwd_);
+                    cwd_buf[cwd_.len] = 0;
+                    const cwd_handle = bun.openDirNoRenamingOrDeletingWindows(bun.invalid_fd, cwd_buf[0..cwd_.len :0]) catch break :brk cwd_;
+
+                    var buf: bun.WPathBuffer = undefined;
+                    const new_cwd = bun.windows.GetFinalPathNameByHandle(cwd_handle.fd, .{}, &buf) catch break :brk cwd_;
+
+                    break :brk strings.convertUTF16toUTF8InBuffer(&cwd_buf, new_cwd) catch break :brk cwd_;
+                };
+
                 return .{
                     .items = scripts,
                     .first_index = @intCast(first_index),
                     .total = total,
-                    .cwd = allocator.dupe(u8, cwd) catch bun.outOfMemory(),
+                    .cwd = allocator.dupeZ(u8, cwd) catch bun.outOfMemory(),
                     .package_name = package_name,
                 };
             }

--- a/test/cli/install/bun-workspaces.test.ts
+++ b/test/cli/install/bun-workspaces.test.ts
@@ -484,3 +484,28 @@ for (const glob of [true, false]) {
     });
   });
 }
+
+test("cwd in workspace script is not the symlink path on windows", async () => {
+  await Promise.all([
+    write(
+      join(packageDir, "package.json"),
+      JSON.stringify({
+        name: "foo",
+        workspaces: ["pkg1"],
+      }),
+    ),
+    write(
+      join(packageDir, "pkg1", "package.json"),
+      JSON.stringify({
+        name: "pkg1",
+        scripts: {
+          postinstall: 'bun -e \'require("fs").writeFileSync("cwd", process.cwd())\'',
+        },
+      }),
+    ),
+  ]);
+
+  await runBunInstall(env, packageDir);
+
+  expect(await file(join(packageDir, "node_modules", "pkg1", "cwd")).text()).toBe(join(packageDir, "pkg1"));
+});

